document/metadata.go

```diff
@@ -0,0 +1,101 @@
+package document
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/loader"
+)
+
+// ParseRegoWithAnnotations parse the rego in the indicated directory
+func ParseRegoWithAnnotations(directory string) (ast.FlatAnnotationsRefSet, error) {
+	// Recursively find all rego files (ignoring test files), starting at the given directory.
+	result, err := loader.NewFileLoader().
+		WithProcessAnnotation(true).
+		Filtered([]string{directory}, func(_ string, info os.FileInfo, _ int) bool {
+			if strings.HasSuffix(info.Name(), "_test.rego") {
+				return true
+			}
+
+			if !info.IsDir() && filepath.Ext(info.Name()) != ".rego" {
+				return true
+			}
+
+			return false
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("filter rego files: %w", err)
+	}
+
+	if _, err := result.Compiler(); err != nil {
+		return nil, fmt.Errorf("compile: %w", err)
+	}
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(result.ParsedModules())
+	as := compiler.GetAnnotationSet().Flatten()
+
+	return as, nil
+}
+
+type Section struct {
```

Comment:
some comments explaining `Section` and its attributes might be helpful as it's the backbone of the generation.